registry/storage/driver/storagedriver.go

```diff
@@ -106,11 +106,11 @@ type FileWriter interface {
 }
 
 // PathRegexp is the regular expression which each file path must match. A
-// file path is absolute, beginning with a path components and may contain
-// more of them separated by slashes, where each path component is
+// file path is absolute, beginning with a slash or a path components and may
+// contain more of them separated by slashes, where each path component is
 // restricted to alphanumeric characters or a period, underscore, colon or
 // hyphen.
-var PathRegexp = regexp.MustCompile(`^([A-Za-z0-9._:-]+(/[A-Za-z0-9._:-]+)*)+$`)
+var PathRegexp = regexp.MustCompile(`^([A-Za-z0-9._:-]*(/[A-Za-z0-9._:-]+)*)+$`)
```

Comment:
we use it upload and download files mostly like https://github.com/deis/builder/blob/master/pkg/gitreceive/build.go#L135
and since we are just trying to store on cloud object storages we don't find any issue with ":" and it can be replaced with a different character in the code if needed.
The main reason why we maintain a fork of the original repo is the ability to create a bucket/container if they don't exist during the init...currently it does for azure and swift but doesn't for the gcs and s3